### PR TITLE
Add slice and onehot

### PR DESCRIPTION
**Pull request by psyyz10**
## What changes were proposed in this pull request?

Add slice and onehot

## How was this patch tested?

Unit tests

### CODE REVIEW

**Comment by yangw1234** [comment]
LGTM

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/OneHot.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.reflect.ClassTag
+
+class OneHot[T: ClassTag](
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/OneHot.scala]: Can you add some documents to explain what's the ops do?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/OneHot.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  axis: Int
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/OneHot.scala]: support onValue/offValue?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/OneHot.scala [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+)(implicit ev: TensorNumeric[T]) extends Operation[Table, Tensor[T], T] {
+  def updateOutput(input: Table): Tensor[T] = {
+    val indices = input[Tensor[Int]](1)
+    val depth = input[Tensor[Int]](2).value()
+    val onValue = input[Tensor[T]](3).valueAt(1)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/OneHot.scala]: why T?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Slice.scala [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+class Slice[T: ClassTag](
+  begin: Array[Int],
+  size: Array[Int])
+  (implicit ev: TensorNumeric[T]) extends Operation[Tensor[T], Tensor[T], T] {
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Slice.scala]: why T

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Slice.scala [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+ */
+package com.intel.analytics.bigdl.nn.ops
+
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+class Slice[T: ClassTag](
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/Slice.scala]: Please add some document for this operation

**Comment by psyyz10** [comment]
@yiheng Please review again

**Comment by yiheng** [comment]
LGTM

**Comment by yiheng** [comment]
PR validation failure http://10.239.47.211:18888/job/BigDL-PR-Validation-1.6-serial/1889/

**Pull request merged**